src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -40,14 +46,49 @@ internal sealed class PowerShellConfig
 
         private PowerShellConfig()
         {
-            // Sets the system-wide configuration directory
-            psHomeConfigDirectory = Utils.DefaultPowerShellAppBase;
+            // Sets the system-wide configuration file.
+            systemWideConfigFile = new FileInfo(Path.Combine(Utils.DefaultPowerShellAppBase, configFileName));
 
             // Sets the per-user configuration directory
             // Note: This directory may or may not exist depending upon the
             // execution scenario. Writes will attempt to create the directory
             // if it does not already exist.
-            appDataConfigDirectory = Utils.GetUserConfigurationDirectory();
+            perUserConfigFile = new FileInfo(Path.Combine(Utils.GetUserConfigurationDirectory(), configFileName));
+        }
+
+        private string GetConfigFilePath(ConfigScope scope)
+        {
+            fileLock.EnterReadLock();
+            try
+            {
+                return (scope == ConfigScope.CurrentUser) ? perUserConfigFile.FullName : systemWideConfigFile.FullName;
+            }
+            finally
+            {
+                fileLock.ExitReadLock();
+            }
+        }
+
+        /// <summary>
+        /// Sets the system wide configuration file path
```

Comment:
Can you please explicitly point out that this method should only be used when parsing the flag `-SettingsPath` for `pwsh` in `CommandLineParameterParser.cs`?

Reply:
Fixed